RecoMuon/TrackerSeedGenerator/plugins/TSGForOI.cc

```diff
@@ -179,10 +192,20 @@ void TSGForOI::produce(edm::StreamID sid, edm::Event& iEvent, const edm::EventSe
       for (auto it=tecNegative.rbegin(); it!=tecNegative.rend(); ++it) {
 	LogTrace("TSGForOI") << "TSGForOI::produce: looping in TEC- layer " << layerCount << endl; 
 	findSeedsOnLayer(tTopo, **it, tsosAtIP,  *(propagatorAlong.get()), *(propagatorOpposite.get()), l2, 
-			 estimatorH, measurementTrackerH, numSeedsMade, numOfMaxSeeds, layerCount, foundHitlessSeed, analysedL2, out);
-      }
+			 estimatorH, measurementTrackerH, numSeedsMade, numOfMaxSeeds, layerCount, analysedL2, out);
+	}
+	if (useBoth){
+		numSeedsMade=0;
+	      	for (auto it=tecNegative.rbegin(); it!=tecNegative.rend(); ++it) {
+			LogTrace("TSGForOI") << "TSGForOI::produce: looping in TEC- layer " << layerCount << endl; 
+			findSeedsOnLayer(tTopo, **it, tsosAtMuonSystem,  *(propagatorOpposite.get()), *(propagatorOpposite.get()), l2, 
+			 estimatorH, measurementTrackerH, numSeedsMade, numOfMaxSeeds, layerCount, analysedL2, out);
+		}
+      	}
     }
 
```

Comment:
the following should apparently replace L148 - L206 and get rid of triple copy-paste.

Capturing of individual variables was unavoidable instead of a simple ```[&]```
```c++
    //generic region lambda
    auto loopOneRegion = [&layerCount,&tTopo,&l2,&estimatorH,&measurementTrackerH,&numSeedsMade,&numOfMaxSeeds,&analysedL2,&out,this]
      (const auto& tsos, const auto& reg, const std::string& regS, const auto& propA, const auto& propB){
      for (auto it=reg.rbegin(); it!=reg.rend(); ++it) {
        LogTrace("TSGForOI") << "TSGForOI::produce: looping in "<<regS<<" layer " << layerCount << endl;
        findSeedsOnLayer(tTopo, **it, tsos,  *propA, *propB, l2,
                         estimatorH, measurementTrackerH, numSeedsMade, numOfMaxSeeds, layerCount, analysedL2, out);
      }
    };
    auto doRegion = [&tsosAtIP,&tsosAtMuonSystem,&propagatorAlong,&propagatorOpposite,&layerCount,&numSeedsMade,&useBoth,&loopOneRegion]
      (const auto& reg, const std::string& regS){
      layerCount = 0;
      loopOneRegion(tsosAtIP, reg, regS, propagatorAlong, propagatorOpposite);
      if (useBoth){
        numSeedsMade=0;
        loopOneRegion(tsosAtMuonSystem, reg, regS, propagatorOpposite, propagatorOpposite);
      }
    };

    //          BARREL
    if (std::abs(l2->eta()) < maxEtaForTOB_) doRegion(tob, "TOB");

    //Reset Number of seeds if in overlap region:
    if (std::abs(l2->eta())>minEtaForTEC_ && std::abs(l2->eta())<maxEtaForTOB_){
      numSeedsMade=0;
    }

    //          ENDCAP+
    if (l2->eta() > minEtaForTEC_)  doRegion(tecPositive, "TEC+");
    //          ENDCAP-
    if (l2->eta() < -minEtaForTEC_)  doRegion(tecNegative, "TEC-");
```

feel free to consider this for refactoring. It compiles, but I didn't cross check for identity.